hal/armv7m/imxrt106x/timer.c

```diff
@@ -14,74 +14,58 @@
  */
 
 #include <hal/hal.h>
-#include <hal/timer.h>
 
 
 enum { gpt_cr, gpt_pr, gpt_sr, gpt_ir, gpt_ocr1, gpt_ocr2, gpt_ocr3, gpt_icr1, gpt_icr2, gpt_cnt };
 
 
-typedef struct {
+struct {
 	volatile u32 *base;
-	volatile u32 done;
-	u32 ticksPerMs;
-
-	u16 irq;
-} timer_t;
-
-
-timer_t timer_common;
+	volatile time_t time;
+} timer_common;
 
 
 static int timer_isr(u16 irq, void *data)
 {
-	timer_t *timer = (timer_t *)data;
-
-	*(timer->base + gpt_sr) = 0x1;
-	*(timer->base + gpt_ir) = 0x1;
-
-	timer_common.done = 1;
+	/* Output Compare 1 Interrupt */
+	if (*(timer_common.base + gpt_sr) & 0x1) {
+		*(timer_common.base + gpt_sr) = 0x1; /* clear status */
+		++timer_common.time;
+	}
 
 	imxrt_dataSyncBarrier();
 
 	return 0;
 }
 
 
-int timer_wait(u32 ms, int flags, volatile u16 *p, u16 v)
+time_t hal_getTime(void)
 {
-	timer_common.done = 0;
-
-	/* Set value that determines when a compare event will be generated */
-	*(timer_common.base + gpt_ocr1) = timer_common.ticksPerMs * ms;
-	*(timer_common.base + gpt_ir) |= 0x1; /* Set OF1IE (Output Compare 1 Interrupt Enable) */
-
-	/* Enable timer*/
-	*(timer_common.base + gpt_cr) |= 1;
+	return timer_common.time;
```

Comment:
This needs to be done in the critical section (with irq disabled).

Reply:
done